src/main/java/org/dependencytrack/tasks/metrics/ProjectMetricsUpdateTask.java

```diff
@@ -69,24 +86,64 @@ private void updateMetrics(final UUID uuid) throws Exception {
-                        // This will happen when a component or its associated project have been deleted after the
-                        // task started. Instead of splurging the log with to-be-expected errors, we just log it
-                        // with DEBUG, and ignore it otherwise.
-                        LOGGER.debug("Couldn't update metrics of component " + component.getUuid() + " because the component was not found", ex);
-                        continue;
-                    } catch (Exception ex) {
-                        LOGGER.error("An unexpected error occurred while updating metrics of component " + component.getUuid(), ex);
-                        continue;
-                    }
+            // if the project is a collection, different logic has to be applied depending on project configuration
+            switch (project.getCollectionLogic()) {
+                case NONE -> this.updateRegularProjectMetrics(project, pm, counters);
+                case AGGREGATE_DIRECT_CHILDREN -> this.updateAggregateDirectChildrenCollectionMetrics(project, pm, counters);
+                case AGGREGATE_DIRECT_CHILDREN_WITH_TAG -> this.updateAggregateDirectChildrenWithTagCollectionMetrics(project, pm, counters);
+                case HIGHEST_SEMVER_CHILD -> this.updateHighestSemVerChildCollectionMetrics(project, pm, counters);
+            }
+
+            qm.runInTransaction(() -> {
+                final ProjectMetrics latestMetrics = qm.getMostRecentProjectMetrics(project);
+                if (!counters.hasChanged(latestMetrics)) {
+                    LOGGER.debug("Metrics of project " + uuid + " did not change");
+                    latestMetrics.setLastOccurrence(counters.measuredAt);
+                } else {
+                    LOGGER.debug("Metrics of project " + uuid + " changed");
+                    final ProjectMetrics metrics = counters.createProjectMetrics(project);
+                    pm.makePersistent(metrics);
+                }
+            });
+
+            if (project.getLastInheritedRiskScore() == null ||
+                    project.getLastInheritedRiskScore() != counters.inheritedRiskScore) {
+                LOGGER.debug("Updating inherited risk score of project " + uuid);
+                qm.runInTransaction(() -> project.setLastInheritedRiskScore(counters.inheritedRiskScore));
+            }
+
+            LOGGER.debug("Completed metrics update for project " + uuid + " in " +
+                    DurationFormatUtils.formatDuration(new Date().getTime() - counters.measuredAt.getTime(), "mm:ss:SS"));
+
+            Project parent = project.getParent();
+            if(parent != null && parent.getCollectionLogic() != ProjectCollectionLogic.NONE) {
+                LOGGER.debug("Scheduling metrics update of project's parent collection " + parent.getUuid());
+                Event.dispatch(new ProjectMetricsUpdateEvent(parent.getUuid()));
+            }
```

Comment:
@nscuro  
I would assume the latest update would then have the latest data, even if running more or less in parallel. If we really would face such issue in rare cases, it would probably fix itself with the next portfolio metric update. or?

Regarding Performance:
What we could do as quick improvement is adding a condition in line 119 to only trigger the parent update when the counters changed. Would need to overthink again if there was a reason why I didn't do this or if I just forgot this.
Usually, the counters should not change too often I would assume, or?

Reply:
I added a check if metrics of current project changed and only trigger an update of parent in this case. At least reduces the calculation effort.

Reply:
The check you added seems to work well in my initial testing. 👍 

However I think a test with a large portfolio would be justified to ensure the impact is indeed limited.